data_model/src/lib.rs

```diff
@@ -158,6 +158,33 @@ impl FromStr for Name {
     }
 }
 
+#[cfg(features = "ffi")]
+pub unsafe extern "C" fn name_from_str(
+    candidate: *const u8,
+    candidate_len: usize,
+    output: *mut *mut Name,
+) -> iroha_ffi::FfiResult {
+    let candidate = core::slice::from_raw_parts(candidate, candidate_len);
+
+    let method_res = match core::str::from_utf8(candidate) {
+        Err(_error) => return iroha_ffi::FfiResult::Utf8Error,
```

Comment:
Question:

Do we plan to have fully typed error enums from inside the FFI? Right now writing WASM is extremely difficult, but at least the core functions return meaningful error messages.

Reply:
yes, [this](https://michael-f-bryan.github.io/rust-ffi-guide/errors/return_types.html) is how it should be done but I figured this would suffice for the purpose of MVP. Error handling should be done in a separate ticket